README.md

```diff
@@ -23,10 +23,24 @@ The 'llama-recipes' repository is a companion to the [Meta Llama 2](https://gith
 >
 > {{ user_message_2 }}<|eot_id|><|start_header_id|>assistant<|end_header_id|>
 > ```
-> More details on the new tokenizer and prompt template: <PLACEHOLDER_URL>
+>
+> To signal the end of the current message the model emits the `<\|eot_id\|>` token. To terminate the generation we need to call the model's generate function as follows:
```

Comment:
message the model => message, the model

Reply:
generation we => generation, we

Reply:
@mreso is it like two  the two EOS terminators should be used to stop generation early? wonder if that. matches our description here or we need a bit of lingo?

Reply:
Is the `eos_token_id` arg specifying the stop sequence for generation?

I think some lingo around understanding the difference between eot_id and end_of_text usage would be helpful

Reply:
Thanks for the comments! Yes, the eos_token_id is the one thats checked in the [stopping criteria](https://github.com/huggingface/transformers/blob/d2cec09baa8e88402ac3da86dcee478ae1446c83/src/transformers/generation/stopping_criteria.py#L132) and usually thats set to [<|end_of_text|>](https://huggingface.co/meta-llama/Meta-Llama-3-8B/blob/b6887ce03ea47d068bf8502ba6ed27f8c5c12a6b/tokenizer_config.json#L12). But for dialog kind of prompt the model is trained to use [<|eot_id|>](https://huggingface.co/meta-llama/Meta-Llama-3-8B/blob/b6887ce03ea47d068bf8502ba6ed27f8c5c12a6b/tokenizer_config.json#L76) (probably to distinguish it from the more final end of sequence). Thats why we need to replace the eos_token_id against the latter id. Otherwise generate rambles on like in this example:
```
Model output:
<|begin_of_text|><|start_header_id|>system<|end_header_id|>

Always answer with emojis<|eot_id|><|start_header_id|>user<|end_header_id|>

How to go from Beijing to NY?<|eot_id|><|start_header_id|>assistant<|end_header_id|>

🛫️🚀🛬<|eot_id|><|start_header_id|>assistant<|end_header_id|>

🏨🛬🇨🇳                                                                                                                                                                                                                                 🕰️ 12+hours o
💺Business Class
[...]
```
The model learned that after an <|eot_id|> comes another header so its adds <|start_header_id|>assistant<|end_header_id|> and then comes another response. (The header is usually appended by the chat template, not the model)
If we exchange the eos_token_id in generate it stops after the model emits the first <|eot_id|>:
```
Model output:
<|begin_of_text|><|start_header_id|>system<|end_header_id|>

Always answer with emojis<|eot_id|><|start_header_id|>user<|end_header_id|>

How to go from Beijing to NY?<|eot_id|><|start_header_id|>assistant<|end_header_id|>

✈️ 🗼️🛬<|eot_id|>
```
Will rework the text accordingly before merigng.

Reply:
So this should have been addressed in this PR which has been merged, https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct/discussions/4/files